core/storage/suggestion/gae_models.py

```diff
@@ -0,0 +1,267 @@
+    """Model to store suggestions made by Oppia users."""
+
+    # The type of suggestion.
+    suggestion_type = ndb.StringProperty(
+        required=True, indexed=True, choices=SUGGESTION_TYPE_CHOICES)
+    # The type of entity which the suggestion is linked to.
+    entity_type = ndb.StringProperty(
+        required=True, indexed=True, choices=ENTITY_TYPE_CHOICES)
+    # Status of the suggestion.
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # Additional parameters to be stored.
+    suggestion_params = ndb.StructuredProperty(SuggestionParams)
+    # The author of the suggestion.
+    author_id = ndb.StringProperty(required=True, indexed=True)
+    # The reviewer who accepted the suggestion.
+    reviewer_id = ndb.StringProperty(required=False, indexed=True)
+    # The thread linked to this suggestion.
+    thread_id = ndb.StringProperty(required=True, indexed=True)
+    # The reviewer assigned to review the suggestion.
+    assigned_reviewer_id = ndb.StringProperty(required=True, indexed=True)
+    # The suggestion payload. Contains the data of the suggestion. For "edit"
+    # suggestions, this contains the change_list. For "add" suggestions, this
+    # contains the data necessary to add the new entity.
+    payload = ndb.JsonProperty(required=True, indexed=False)
+
+    @classmethod
+    def get_instance_id(
+            cls, entity_type, thread_id, entity_id):
+        """Concatenates various parameters and gives the ID of the suggestion
+        model.
+
+        Args:
+            entity_type: str. The type of entity being edited/added.
+            thread_id: str. The ID of the feedback thread linked to the
+                suggestion.
+            entity_id: str. The ID of the entity being edited.
+
+        Returns:
+            str. The full instance ID for the suggestion.
+        """
+        return SUGGESTION_ID_DELIMITER.join([entity_type, thread_id, entity_id])
+
+    @classmethod
+    def create(
+            cls, suggestion_type, entity_type, status,
+            suggestion_params, author_id, reviewer_id, thread_id,
+            assigned_reviewer_id, payload):
+        """Creates a new SuggestionModel entry.
+
+        Args:
+            suggestion_type: str. The type of the suggestion.
+            entity_type: str. The type of entity being edited/added.
+            status: str. The status of the suggestion.
+            suggestion_params: dict. Additional parameters for the suggestion
+            author_id: str. The ID of the user who submitted the suggestion.
+            reviewer_id: str. The ID of the reviewer who has accepted the
+                suggestion.
+            thread_id: str. The ID of the feedback thread linked to the
+                suggestion.
+            assigned_reviewer_id: str. The ID of the user assigned to
+                review the suggestion.
+            payload: dict. The actual content of the suggestion.
```

Comment:
shouldn't this be a list of dicts, each representing a changelist cmd? I'm not sure -- please double check.

Reply:
Ah, thanks for the clarification. Then maybe call it change_cmd?

Reply:
Done.